File: app/src/main/java/com/urban/androidhomework/UrbanHomeworkApp.java
package com.urban.androidhomework;

import android.app.Application;

import com.urban.androidhomework.api.NetworkClient;

public class UrbanHomeworkApp extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        NetworkClient.get().setup(getApplicationContext());
    }
}

File: app/src/main/java/com/urban/androidhomework/UrbanHomeworkApp.kt
package com.urban.androidhomework

import android.app.Application
import dagger.hilt.android.HiltAndroidApp
import timber.log.Timber

@HiltAndroidApp
class UrbanHomeworkApp : Application() {
    override fun onCreate() {
        super.onCreate()
        initialiseApp()
    }

    private fun initialiseApp() {
        // Setup logger
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
    }
}